insights/specs/datasources/pcp.py

```diff
@@ -110,12 +110,20 @@ def pcp_raw_files(broker):
         ContentException: Raises when no such directory or number of PCP RAW
                           files less than 3.
     """
-    hn = broker[Hostname]
-    pm_root = os.path.join("/var/log/pcp/pmlogger/", hn.hostname)
-    if not os.path.exists(pm_root):
-        pm_root = os.path.join("/var/log/pcp/pmlogger/", hn.fqdn)
-    if not os.path.exists(pm_root):
-        raise ContentException("No PCP data directory found for {0} or {1}".format(hn.hostname, hn.fqdn))
+    hostnames = set()
+    hnd = broker.get(HostnameDefault)
+    hostnames.add(hnd.raw) if hnd else None
+    hn = broker.get(Hostname)
+    hostnames.update([hn.hostname, hn.fqdn]) if hn else None
+
+    pm_root = None
+    for hostname in hostnames:
+        tmp_root = os.path.join("/var/log/pcp/pmlogger/", hostname)
+        if os.path.exists(tmp_root):
+            pm_root = tmp_root
+            break
```

Comment:
Adding for notes:
There could be situation where the first directory we find does not have pcp archives of yesterday and hence we may never report any files from that system, because everytime the collector would run it would default to same directory.

So we need to search in every directory named by shortname, fqdn or raw to look for yesterday's files. There could be further situations,
1. Two or more direcotries having yesterday's files and both directories are having complete sets(.index, .meta, 0.xz). In this case we are going to upload the files from directory which are latest. The latest timestamp would be of .index file.
2. Two or more directories having yesterday's files however only one of that is having valid set. In that case we upload from the directory which has valid set.
3. Multiple directories having yesterday's files, and some of those are having valid set. In this case we take valid sets, and upload the latest timestamp.